Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  NotificationService.swift
+//  NotificationServiceExtension
+//
+//  Created by Ajay Subramanya on 6/14/23.
+//
+
+import UIKit
+import UserNotifications
+
+// MARK: notification service extension implementation.
+
+/// When push payload is marked as there being mutable-content this service
+/// (more specifically the `didReceiveNotificationRequest` ) is called to perform
+/// tasks such as downloading images and attaching it to the notification before it's displayed to the user.
+///
+/// There is a limited time before which `didReceiveNotificationRequest`  needs to wrap up it's operations
+/// else the notification is displayed as received.
+///
+/// Any property from `UNMutableNotificationContent` can be mutated here before presenting the notification.
+class NotificationService: UNNotificationServiceExtension {
```

Comment:
Just curious does this code change much for gif/video? Or will this work once the server supports sending those?

Reply:
Good call. I just updated the code to work in  a media agnostic way. Passing down the media type so we don't hardcode it to png but instead use what app sends and if it doesn't send any we default to png.